### PR TITLE
Add GeoAreas tab to Project page

### LINKED ISSUE #1781: New project flow: Create geo-areas tab
### New Addition
* Setup page layout
* Fetch list of geo-areas of the project (only fetch, no need to show on map for now) and show it on the right pane

### Attachments
- [Figma Link](https://www.figma.com/proto/HaQd4COVlpSMoqSfgL5ZDx/DEEP_UI_LIBRARY-%5Bin-progress%5D?node-id=8700%3A83234&scaling=min-zoom)

### LINKED ISSUE #1782: New project flow: List custom geo-areas on right pane
### New Addition
* If region is private and created for the project, make region expandable to list region details
    * Title
    * Geo area code
    * List of admin levels in tabs
* No need to make the form submitabble in this issue

### Attachments
- [Figma Link](https://www.figma.com/proto/HaQd4COVlpSMoqSfgL5ZDx/DEEP_UI_LIBRARY-%5Bin-progress%5D?node-id=8700%3A83234&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by samshara**
- fixes https://github.com/the-deep/client/issues/1781
-  fixes https://github.com/the-deep/client/issues/1782

## Changes

* Add Geo Areas tab to project page
* Display list of geo-areas of project
* Display details of geo-areas that are private to the project

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    className?: string;
+}
+
+function RegionCard(props: Props) {
+    const {
+        className,
+        region,
+    } = props;
+
+    const handleDeleteUserGroupClick = () => {}; // FIXME  this will be added later
```
Comment (AdityaKhatri) [inline src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx]: Change name as there is no usergroup here

File: src/newViews/ProjectEdit/GeoAreas/index.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+import {
+    BasicRegion,
+} from '#typings';
+import _ts from '#ts';
+
+import RegionCard from './RegionCard';
+import styles from './styles.scss';
+
+const regionKeySelector = (d: BasicRegion): string => d.id.toString();
```
Comment (AdityaKhatri) [inline src/newViews/ProjectEdit/GeoAreas/index.tsx]: What is toString() necessary?

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/newViews/ProjectEdit/GeoAreas/index.tsx (hunk 2 above)
Comment (samshara) [reply src/newViews/ProjectEdit/GeoAreas/index.tsx]: @AdityaKhatri the ListView keySelector doesn't take number as key

Comment (tnagorra) [reply src/newViews/ProjectEdit/GeoAreas/index.tsx]: That is not true.
The problem here is that the regionRendererParams has defined key to be string 
```
    const regionRendererParams = useCallback((_: string, data: BasicRegion) => ({
```

**Review by tnagorra — APPROVED** [review]

File: src/newViews/ProjectEdit/GeoAreas/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                {showContent && (
+                    <ListView
+                        className={styles.regions}
+                        data={regions}
+                        rendererParams={regionRendererParams}
+                        renderer={RegionCard}
+                        rendererClassName={styles.region}
+                        keySelector={regionKeySelector}
+                    />
+                )}
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/index.tsx]: We already have a component to do this.
Also, the default state should not be collapsed right @AdityaKhatri ?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/newViews/ProjectEdit/GeoAreas/index.tsx (hunk 3 above)
Comment (samshara) [reply src/newViews/ProjectEdit/GeoAreas/index.tsx]: I have made the necessary changes

File: src/newViews/ProjectEdit/GeoAreas/index.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+import {
+    BasicRegion,
+} from '#typings';
+import _ts from '#ts';
+
+import RegionCard from './RegionCard';
+import styles from './styles.scss';
+
+const regionKeySelector = (d: BasicRegion): number => d.id;
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/index.tsx]:
```suggestion
const regionKeySelector = (d: BasicRegion) => d.id;
```

File: src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                        onConfirm={handleDeleteRegionClick}
+                        message={_ts('geoAreas', 'deleteGeoAreaConfirm')}
+                        showConfirmationInitially={false}
+                    >
+                        <IoTrash />
+                    </QuickActionConfirmButton>
+                </>
+            )}
+        >
+            { (response && !response.public && response.adminLevels.length > 0) && (
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx]:
```suggestion
            {(response && !response.public && response.adminLevels.length > 0) && (
```

File: src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                    <QuickActionConfirmButton
+                        name="deleteButton"
+                        title={_ts('geoAreas', 'deleteGeoArea')}
+                        onConfirm={handleDeleteRegionClick}
+                        message={_ts('geoAreas', 'deleteGeoAreaConfirm')}
+                        showConfirmationInitially={false}
+                    >
+                        <IoTrash />
+                    </QuickActionConfirmButton>
+                </>
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx]:
```suggestion
                    <QuickActionConfirmButton
                        name="deleteButton"
                        title={_ts('geoAreas', 'deleteGeoArea')}
                        onConfirm={handleDeleteRegionClick}
                        message={_ts('geoAreas', 'deleteGeoAreaConfirm')}
                        showConfirmationInitially={false}
                    >
                        <IoTrash />
                    </QuickActionConfirmButton>
```

File: src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+import {
+    AdminLevel,
+    BasicRegion,
+    Region,
+} from '#typings';
+
+import AdminLevelCard from './AdminLevelCard';
+import styles from './styles.scss';
+
+const adminLevelKeySelector = (d: AdminLevel): number => d.id;
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/RegionCard/index.tsx]:
```suggestion
const adminLevelKeySelector = (d: AdminLevel) => d.id;
```

**Pull request merged**